average.js

```diff
@@ -1,6 +1,20 @@
-
 function average(numbers) {
-  return numbers.reduce((p, c)=> p + c, 0) / numbers.length;
-}
-
+    if(!numbers?.length)
+    {
+        return NaN;
+    }
+    
+    let count=1;
+    let sumTotal = numbers.reduce((p, c) => {
```

Comment:
You are doing the summing and elimination of NaN in a single pass. This is good in many situations.
However, consider the trade-off as well: Making the summing and counting NaNs in separate loops enables them to run in parallel, resulting in faster turnaround-time for large data-sets